test/e2e/e2e_test.go

```diff
@@ -95,6 +96,15 @@ var _ = Describe("etcd-operator", Ordered, func() {
 				ExpectWithOffset(1, err).NotTo(HaveOccurred())
 			})
 
+			Eventually(func() error {
+				cmd := exec.Command("kubectl", "get",
+					"statefulset/test",
+					"--namespace", namespace,
+				)
+				_, err = utils.Run(cmd)
+				return err
+			}, time.Second*20, time.Second*2).Should(Succeed())
+
 			By("wait for statefulset is ready", func() {
```

Comment:
Eventually + By can be reduced to a single Eventually with kubectl wait. That will be terser.